include expression/extent in filter_put expression template, include perfDuration in HTML output; fixes #158

peterprovaznik opened the pull request:
"extent" was not part of the list of apply-templates instructions in the expression template, which resulted in it being removed post-`filter_put.xsl`.

Clemens (@gucl-mu) insists (and I tend to agree) that extent should really have both `@quantity` and `@unit` (which would require the list of supported units in `unit-select.xml`) to be useful. This, however, would be a non-trivial change in several places which in my opinion would unreasonably widen the scope of this PR (and should have a PR of its own).

I have also noticed that "height" is hard-coded in the `xf:setvalue` in `unit-select.xml`, which is most likely erroneous (since this is also used for e.g. width); I have, however, since this is not directly related to the issue at hand, decided not to change it here (originally I did, then tests failed, then I reverted it, tests still failed, then I realized it should most likely be handled separately).

peterstadler commented:
Hi @peterprovaznik , could you please rebase your branch on develop and force commit? There have been some updates to the test dependencies #180 which should fix the failing tests here.

peterstadler commented:
@daniel-jettka and I were just having a look and found that
* `<extent>` was saved to XML and displayed as HTML properly 
* `<perfDuration>` was saved to XML and displayed as HTML properly **WHEN** it was entered on the highest (i.e. work level?) expression 

* `<perfDuration>` was saved to XML **BUT NOT** displayed as HTML properly when entered on the component level (e.g. `expression/componentList/expression/perfDuration`)

peterstadler reviewed (changes requested):
@daniel-jettka and I were just having a look and found that
* `<extent>` was saved to XML and displayed as HTML properly 
* `<perfDuration>` was saved to XML and displayed as HTML properly **WHEN** it was entered on the highest (i.e. work level?) expression 

* `<perfDuration>` was saved to XML **BUT NOT** displayed as HTML properly when entered on the component level (e.g. `expression/componentList/expression/perfDuration`)

peterprovaznik commented:
Thanks for reviewing! I have added the `<apply-templates/>` for `<perfDuration>` in the `<componentList>/<expression>` template as well. I have also reverted the added newlines to reduce diff-noise, so feel free to squash when merging.
